src-tauri/src/binary_resolver.rs

```diff
@@ -1,516 +0,0 @@
-use crate::download_utils::{download_file_with_retries, extract, validate_checksum};
-use crate::{github, ProgressTracker};
-use anyhow::{anyhow, Error};
-use async_trait::async_trait;
-use log::{debug, error, info, warn};
-use regex::Regex;
-use semver::{Version, VersionReq};
-use std::collections::HashMap;
-use std::path::PathBuf;
-use std::sync::{Arc, LazyLock};
-use tari_common::configuration::Network;
-use tauri::api::path::cache_dir;
-use tokio::fs;
-use tokio::sync::{Mutex, RwLock};
-
-const LOG_TARGET: &str = "tari::universe::binary_resolver";
-static INSTANCE: LazyLock<BinaryResolver> = LazyLock::new(BinaryResolver::new);
-
-pub struct BinaryResolver {
-    download_mutex: Mutex<()>,
-    adapters: HashMap<Binaries, Box<dyn LatestVersionApiAdapter>>,
-    latest_versions: Arc<RwLock<HashMap<Binaries, Version>>>,
-}
-
-#[derive(Debug, Clone)]
-pub struct VersionDownloadInfo {
-    pub(crate) version: Version,
-    pub(crate) assets: Vec<VersionAsset>,
-}
-
-#[derive(Debug, Clone)]
-pub struct VersionAsset {
-    pub(crate) url: String,
-    pub(crate) name: String,
-}
-
-#[async_trait]
-pub trait LatestVersionApiAdapter: Send + Sync + 'static {
-    async fn fetch_latest_release(&self) -> Result<VersionDownloadInfo, Error>;
-    fn is_version_allowed(&self, version: &Version) -> bool;
-
-    fn get_binary_folder(&self) -> PathBuf;
-
-    fn find_version_for_platform(
-        &self,
-        version: &VersionDownloadInfo,
-    ) -> Result<VersionAsset, Error>;
-}
-
-pub struct XmrigVersionApiAdapter {}
-
-#[async_trait]
-impl LatestVersionApiAdapter for XmrigVersionApiAdapter {
-    async fn fetch_latest_release(&self) -> Result<VersionDownloadInfo, Error> {
-        todo!()
-    }
-
-    fn is_version_allowed(&self, _version: &Version) -> bool {
-        todo!()
-    }
-
-    fn get_binary_folder(&self) -> PathBuf {
-        todo!()
-    }
-
-    fn find_version_for_platform(
-        &self,
-        _version: &VersionDownloadInfo,
-    ) -> Result<VersionAsset, Error> {
-        todo!()
-    }
-}
-
-pub struct GithubReleasesAdapter {
-    pub repo: String,
-    pub owner: String,
-    pub semver: Option<VersionReq>,
-    pub version_pre_filter: Option<Regex>,
```

Comment:
These fields are missing in your new file

Reply:
Hey, they are included in binary_manager